### ISSUE TITLE
How do you change the day color dot

### ISSUE THREAD

**Issue by justdan0227** [kvyatkovskys/KVKCalendar#62]
I want to change the background color of the day (month view) based on the events in that day.  Is this possible?

**Comment by kvyatkovskys** [comment]
It's possible `MonthStyle -> colorBackgroundDate, colorBackgroundWeekendDate`

**Comment by justdan0227** [comment]
Sorry I'm lost.. If I am supposed to have 4 events for a day and only 3 have been set, how do I change the background color of that day and then change it once the 4th event has been set?

**Comment by kvyatkovskys** [comment]
🤔interesting ....
perhaps, need to add a new method to change the color of cell (for this case)

**Comment by justdan0227** [comment]
That would be awesome!

**Comment by justdan0227** [comment]
Any update?

**Comment by kvyatkovskys** [comment]
yep, added a method `func willDisplayDate(_ date: Date?, events: [Event]) -> DateStyle?` to protocol `CalendarDataSource`. Need to be updated to v0.2.9. 

for example... call the method from your controller
`extension ViewController: CalendarDataSource { }`

**Comment by justdan0227** [comment]
awesome!  Trying it out now!

**Comment by justdan0227** [comment]
So this is great (works in Month but not in Week/Day - can it?).  

Also this changes the background color. I just want to put a  colored circle behind the date like it does with the current date (red).

**Comment by kvyatkovskys** [comment]
🙂 I’m working it.

**Comment by justdan0227** [comment]
any update.. anything I can help with?

**Comment by kvyatkovskys** [comment]
I plan to finish tomorrow morning

**Comment by justdan0227** [comment]
you rock!

**Comment by kvyatkovskys** [comment]
Released a new version 0.3.0. 🙂

**Comment by justdan0227** [comment]
My app was crashing when I went to Month View.  Turns out I had to clean build folder and rebuild.  No longer crahses, however I have it turning all of the days dots to yellow for April, and only one is, and its random.  

import SwiftDate

`    func willDisplayDate(_ date: Date?, events: [Event]) -> DateStyle? {

        if date == nil {
            return nil
        }

        let startDate = "2020-04-01 15:30:00".toDate()
        let endDate   = "2020-04-30 15:30:00".toDate()
        
        if date!.isInRange(date: startDate!.date, and: endDate!.date) {
            let dateStyle = DateStyle(backgroundColor: .white,
                                      textColor: .black,
                                      dotBackgroundColor: .yellow)
            return dateStyle
        }

        return nil

    }`

**Comment by kvyatkovskys** [comment]
need add the method `func willDisplayDate(_ date: Date?, events: [Event]) -> DateStyle?` from `CalendarDataSource` and that is all.
Can you send a crash log?

**Comment by justdan0227** [comment]
The calendar at the top of day and week view do not appear to be affected by the coloring?

**Comment by justdan0227** [comment]
Should be all yellow dots

![Screen Shot 2020-05-02 at 1 46 37 PM](https://user-images.githubusercontent.com/918740/80873021-61544480-8c7b-11ea-9eb9-da263d1303ed.png)

**Comment by kvyatkovskys** [comment]
Example project works fine🤔. 
Perhaps, you have a problem in condition `if date!.isInRange(date: startDate!.date, and: endDate!.date)`

**Comment by kvyatkovskys** [comment]
😆 I added a new code in wrong place)
sorry, need to rewrite

**Comment by justdan0227** [comment]
K I just tried this and only today is turning yellow: :)

 func willDisplayDate(_ date: Date?, events: [Event]) -> DateStyle? {
            let dateStyle = DateStyle(backgroundColor: .white,
                                      textColor: .black,
                                      dotBackgroundColor: .yellow)
            return dateStyle
    }

Thanks!

**Comment by justdan0227** [comment]
Any progress? :)

**Comment by kvyatkovskys** [comment]
The next version will be released on weekends

**Comment by kvyatkovskys** [comment]
Released a new v0.3.1. You can try it now.

**Comment by justdan0227** [comment]
thanks.. trying it now!

**Comment by justdan0227** [comment]
Hmmm what am I missing?  I have the following and should get all of the days in April to show up as green and yet only the current day does? (Using SwiftDate to create the dates).  The log shows the date is printed, however I only get one green dot.

` 
let startDate = "2020-04-01 15:30:00".toDate()
        let endDate   = "2020-04-30 15:30:00".toDate()
        
        //if date!.isInRange(date: startDate!.date, and: endDate!.date) {
        if (date! < endDate!.date)  && (date! > startDate!.date) {
            let dateStyle = DateStyle(backgroundColor: .white,
                                      textColor: .black,
                                      dotBackgroundColor: .green)
            print (date!.description)
            return dateStyle
        }

        return nil
`

**Comment by justdan0227** [comment]
in fact shouldn't this just make every day a green circle?

```
func willDisplayDate(_ date: Date?, events: [Event]) -> DateStyle? {

    let dateStyle = DateStyle(backgroundColor: .white,
                              textColor: .black,
                              dotBackgroundColor: .green)
    print (date!.description)
    return dateStyle
    }
```

**Comment by kvyatkovskys** [comment]
> in fact shouldn't this just make every day a green circle?
> 
> ```
> func willDisplayDate(_ date: Date?, events: [Event]) -> DateStyle? {
> 
>     let dateStyle = DateStyle(backgroundColor: .white,
>                               textColor: .black,
>                               dotBackgroundColor: .green)
>     print (date!.description)
>     return dateStyle
>     }
> ```

this should make🤔
let me check....

**Comment by kvyatkovskys** [comment]
try it like this `DateStyle(backgroundColor: .green, textColor: .black, dotBackgroundColor: .green)`
<img width="431" alt="Screenshot 2020-05-11 at 21 04 26" src="https://user-images.githubusercontent.com/8233076/81595342-01704300-93cb-11ea-974b-6b0fd7b92bf1.png">

**Comment by justdan0227** [comment]
so... that "works.".. however, I guess I'm not understanding the "dot" vs background color? (btw .. this ROCKS!)

**Comment by justdan0227** [comment]
So in day and week, setting the background color changes the circle to the desired color.  However, in month view it changes the background square of the day to the color vs just the circle.

Thoughts?

**Comment by justdan0227** [comment]
Can you update the doc on this new functionality?  I can't figure out how to set the color for the circle for the selected day, and what to set the color for events.  Not sure what background color does vs dot color.  Thanks!

**Comment by kvyatkovskys** [comment]
> So in day and week, setting the background color changes the circle to the desired color.  However, in month view it changes the background square of the day to the color vs just the circle.
> 
> 
> 
> Thoughts?
> 
> 

In the next version I want to refactor the code a bit. And I’ll update the readme file.

**Comment by justdan0227** [comment]
So in the end,  I would like to be able to 
set Current Day to a blue circle white number
set default days to be white circle black numbers
set an event day to orange circle black number
set a day that needs to be reviewed as red circle and yellow.

Thoughts?

**Comment by justdan0227** [comment]
Any progress?

**Comment by kvyatkovskys** [comment]
Available style from `HeaderScrollStyle` 

`public var backgroundColor
  public var colorTitleDate
  public var colorTitleCornerDate
  public var colorDate
  public var colorNameDay
  public var colorCurrentDate
  public var colorBackgroundCurrentDate
  public var colorBackgroundSelectDate
  public var colorSelectDate
  public var colorWeekendDate`

**Comment by justdan0227** [comment]
So Week/Day I think are working, however month is not.

So I may not understand    func willDisplayDate(_ date: Date?, events: [Event]) -> DateStyle?  This gets called in month view when I click on a date for every date in the calendar?  Also when I click on the 20th (right now it’s the 19th), both the 19th and 20'th have red circles?
![Screen Shot 2020-05-19 at 12 30 59 PM](https://user-images.githubusercontent.com/918740/82359007-f15cf280-99cc-11ea-8251-4ba59412177d.png)

**Comment by justdan0227** [comment]
Any update?

**Comment by kvyatkovskys** [comment]
> Any update?

Sorry, I have a lot of work. 
What parameters do you use in `willDispayDate`?

**Comment by justdan0227** [comment]
Thank you for all that you do on this library. It really is a great enhancement!    This is what my willDisplayDate looks like today:  (Notice that the selected day has red as well as current day)
[cal.txt](https://github.com/kvyatkovskys/KVKCalendar/files/4698260/cal.txt)

![Screen Shot 2020-05-28 at 3 59 37 PM](https://user-images.githubusercontent.com/918740/83193636-03414280-a0fd-11ea-80dd-68756920b463.png)

**Comment by kvyatkovskys** [comment]
> Thank you for all that you do on this library. It really is a great enhancement! This is what my willDisplayDate looks like today: (Notice that the selected day has red as well as current day)
> [cal.txt](https://github.com/kvyatkovskys/KVKCalendar/files/4698260/cal.txt)
> 
> ![Screen Shot 2020-05-28 at 3 59 37 PM](https://user-images.githubusercontent.com/918740/83193636-03414280-a0fd-11ea-80dd-68756920b463.png)

Thx)

I'll check and fix it. I guess, the next release is on the weekend.

**Comment by justdan0227** [comment]
I don't see a release?

**Comment by kvyatkovskys** [comment]
> I don't see a release?

Release will be today.

**Comment by kvyatkovskys** [comment]
Released a version 0.3.2

**Comment by justdan0227** [comment]
Thanks for the update.  Will you also update the documentation to explain willDisplayDate ?
